public/app/core/logs_model.ts

```diff
@@ -329,10 +329,11 @@ function getLabelsForFrameRow(fields: LogFields, index: number): Labels {
 
   const { stringField, labelsField } = fields;
 
-  return {
-    ...stringField.labels,
-    ...labelsField?.values.get(index),
-  };
+  if (labelsField !== undefined) {
+    return labelsField.values.get(index);
+  } else {
+    return stringField.labels ?? {};
+  }
```

Comment:
i made the code stricter. if your data has the labels-field (which gets found only if you have the specific-frame-type), we only use that source of information